### PR TITLE
Rename argnum with argnums

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**
We use the autograd argnums instead of jax argnums. (grad, value_and_grad, vjp, jvp)

**Description of the Change:**
Argnum is renamed argnums.

### CODE REVIEW

**Comment by rmoyard** [comment]
Closes https://github.com/PennyLaneAI/catalyst/issues/850

**Comment by rmoyard** [comment]
Changes in PL: https://github.com/PennyLaneAI/pennylane/pull/6117

File: benchmark/catalyst_benchmark/measurements.py [hunk 1]
```diff
@@ -493,7 +493,7 @@ def measure_compile_pennylane(a: ParsedArguments) -> BenchmarkResult:
 
         p = Problem(
             qml.device(device, wires=a.nqubits),
-            grad=partial(qml.grad, argnum=0),
+            grad=partial(qml.grad, argnums=0),
```
Comment (rmoyard) [inline benchmark/catalyst_benchmark/measurements.py:496]:
```suggestion
            grad=partial(qml.grad, argnum=0),
```

File: benchmark/catalyst_benchmark/measurements.py [hunk 2]
```diff
@@ -511,7 +511,7 @@ def measure_compile_pennylane(a: ParsedArguments) -> BenchmarkResult:
...
```
Comment (rmoyard) [inline benchmark/catalyst_benchmark/measurements.py:514]:
```suggestion
            grad=partial(qml.grad, argnum=0),
```

File: benchmark/catalyst_benchmark/measurements.py [hunk 3]
```diff
@@ -574,7 +574,7 @@ def measure_runtime_pennylane(a: ParsedArguments) -> BenchmarkResult:
...
```
Comment (rmoyard) [inline benchmark/catalyst_benchmark/measurements.py:577]:
```suggestion
            grad=partial(qml.grad, argnum=0),
```

File: benchmark/catalyst_benchmark/measurements.py [hunk 4]
```diff
@@ -591,7 +591,7 @@ def measure_runtime_pennylane(a: ParsedArguments) -> BenchmarkResult:
...
```
Comment (rmoyard) [inline benchmark/catalyst_benchmark/measurements.py:594]:
```suggestion
            grad=partial(qml.grad, argnum=0),
```

File: benchmark/catalyst_benchmark/test_cases/chemvqe_pennylane.py [hunk 5]
```diff
@@ -136,7 +136,7 @@ def run_default_qubit(N=6):
     """Test problem entry point"""
     p = ProblemCVQE(
         qml.device("default.qubit", wires=N, shots=SHOTS),
-        grad=partial(qml.grad, argnum=0),
+        grad=partial(qml.grad, argnums=0),
```
Comment (rmoyard) [inline benchmark/catalyst_benchmark/test_cases/chemvqe_pennylane.py:139]:
```suggestion
        grad=partial(qml.grad, argnum=0),
```

File: benchmark/catalyst_benchmark/test_cases/chemvqe_pennylane.py [hunk 6]
```diff
@@ -154,7 +154,7 @@ def run_lightning_qubit(N=6):
     """Test problem entry point"""
     p = ProblemCVQE(
         qml.device("lightning.qubit", wires=N, shots=SHOTS),
-        grad=partial(qml.grad, argnum=0),
+        grad=partial(qml.grad, argnums=0),
```
Comment (rmoyard) [inline benchmark/catalyst_benchmark/test_cases/chemvqe_pennylane.py:157]:
```suggestion
        grad=partial(qml.grad, argnum=0),
```

File: demos/adaptive_circuits_demo.ipynb [hunk 7]
```diff
@@ -617,7 +617,7 @@
     "cost_fn = qml.QNode(circuit_1, dev, diff_method=\"adjoint\")\n",
     "cost_fn = decompose(stopping_condition, cost_fn)\n",
     "\n",
-    "circuit_gradient = qml.grad(cost_fn, argnum=0)\n",
+    "circuit_gradient = qml.grad(cost_fn, argnums=0)\n",
```
Comment (rmoyard) [inline demos/adaptive_circuits_demo.ipynb:620]:
```suggestion
    "circuit_gradient = qml.grad(cost_fn, argnum=0)\n",
```

File: demos/adaptive_circuits_demo.ipynb [hunk 8]
```diff
@@ -750,7 +750,7 @@
     "\n",
     "cost_fn = qml.QNode(circuit_2, dev, diff_method=\"adjoint\")\n",
     "cost_fn = decompose(stopping_condition, cost_fn)\n",
-    "circuit_gradient = qml.grad(cost_fn, argnum=0)\n",
+    "circuit_gradient = qml.grad(cost_fn, argnums=0)\n",
```
Comment (rmoyard) [inline demos/adaptive_circuits_demo.ipynb:753]:
```suggestion
    "circuit_gradient = qml.grad(cost_fn, argnum=0)\n",
```

**Review by erick-xanadu — APPROVED** [review]
It looks great. I looked for uses of `argnum` and only the ones from `qml.grad` remain. We agreed offline that the final call for this will be given by @josh146 . Adding here for context this related PR in PennyLane repo: https://github.com/PennyLaneAI/pennylane/pull/6117

Thanks!

File: doc/changelog.md [hunk 9]
```diff
@@ -324,6 +324,9 @@
 
 <h3>Breaking changes</h3>
 
+* The keyword argument argnum for grad, jacobian, vjp and jvp is replaced by argnums.
+  [(#1036)](https://github.com/PennyLaneAI/catalyst/pull/1036)
```
Comment (josh146) [inline doc/changelog.md:328]:
```suggestion
* The `argnum` keyword argument in the `grad`, `jacobian`, `value_and_grad`,
  `vjp`, and `jvp` functions has been renamed to `argnums` to better match JAX.
  [(#1036)](https://github.com/PennyLaneAI/catalyst/pull/1036)
```

**Review by josh146 — APPROVED** [review]
Thanks Romain!

**Pull request merged**